ir/src/tests/variables.rs

```diff
@@ -223,14 +211,13 @@ fn err_ic_variable_access_before_declaration() {
         enf clk' = clk + a
         let a = 1";
 
-    let parsed = parse(source).expect("Parsing failed");
-    let result = AirIR::new(parsed);
-    assert!(result.is_err());
+    expect_diagnostic(source, "expected one of: '\"enf\"', '\"let\"'");
```

Comment:
This error is generated by LALRPOP, not us, so we don't really control errors of this type (this is one of the reasons why hand-written recursive descent parsers are preferable to parser generators, but I digress). We _may_ be able to modify the grammar to allow catching this specific case but I'd have to experiment a bit. 

IIRC the reason why we're leaning on the parser to catch this is due to a combination of factors:

* Statement blocks cannot be empty without making the grammar ambiguous due to the lack of block delimiters (e.g. `{`/`}`),
* If we have a grammar rule to catch a dangling let (e.g. just the `let x = expr` part), then a parse error in what would have been the body of that `let` expression will produce even worse diagnostic output. Specifically, it will fail to parse the "correct" non-terminal rule for `let` expressions, and fall back to the dangling-let rule, which will succeed because that part of the expression is valid. This will produce a dangling-let error even though there isn't one. If we make that type of error a "soft" error, parsing will fail immediately after that on the next token anyway, with an unexpected token error, as it is highly unlikely that the next token will be valid at that point in the parse tree.

The typical solution here is to relax the grammar and catch validation errors during semantic analysis, but that is only possible when relaxing the grammar doesn't make it ambiguous, which I believe is the issue here.